torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1034,6 +1044,36 @@ def sample_inputs_masked_reduction(op_info, device, dtype, requires_grad, **kwar
     return inputs
 
 
+def sample_inputs_mixed_masked_reduction(op_info, device, dtype, requires_grad, **kwargs):
```

Comment:
Instead of generating "mixed" sample inputs from OpInfos I would just model this as a generator specific to test. We can think of mixed layout testing as analogous to mixed dtype testing. OpInfo sample inputs typically don't test type promotion but we have specific tests which do. For example: 

https://github.com/pytorch/pytorch/blob/5e6f29661254e5ebc97fac3d829e5d455cda5864/test/test_binary_ufuncs.py#L566

The reason I'd suggest that approach is because OpInfos are meant to simplify the cross-product of what can be tested, so we don't generally want OpInfos to be worried about generating things like a cross-product of dtypes or layouts.